### ISSUE TITLE
[rush] "rush deploy" should copy PNPM workaround links

### ISSUE THREAD

**Issue by arikon** [microsoft/rushstack#1942]
## Please prefix the issue title with the project name i.e. [rush], [api-extractor] etc. ##

**Is this a feature or a bug?**

- [x] Feature
- [ ] Bug

**Please describe the actual behavior.**

For now `rush deploy` copies only files that a directly pointed to by project dependencies. In case of `pnpm` those files are: `common/temp/node_modules/.pnpm/**/node_modules/*`

`pnpm` also creates symlinks to those files in this location: `common/temp/node_modules/*`

They are useful in case of when some dependency expects another as a peer in the project, but does not directly depend on it.

`rush deploy` could copy all of that symlinks, that point to direct dependencies of the deployable projects.

**If the issue is a bug, how can we reproduce it? Please provide detailed steps and include a GitHub branch if applicable. Your issue will get resolved faster if you can make it easy to investigate.**

**What is the expected behavior?**

**If this is a bug, please provide the tool version, Node.js version, and OS.**

* **Tool:** 
* **Tool Version:** 
* **Node Version:** 
  * **Is this a LTS version?** 
  * **Have you tested on a LTS version?** 
* **OS:**

**Comment by arikon** [comment]
This workaround demonstrates the result I want to get

```sh
# copy all of the symlinks in `common/temp/node_modules/` (but not in the `.pnpm` subfolder)
cp -R common/temp/node_modules/* common/deploy/common/temp/node_modules

cd common/deploy/common/temp/node_modules

# remove all the broken symlinks
find . -path .pnpm -prune -o -type l -exec sh -c 'for x; do [ -e "$x" ] || rm "$x"; done' _ {} +

# remove all the empty directories
find . -path .pnpm -prune -o -type d -empty -delete
```

**Comment by octogonz** [comment]
Thanks for explaining. You are referring to the extra symlinks that PNPM 4 introduced as a broad workaround for any legacy packages that still have phantom dependency problems.

Agreed, we should add this feature. Note that it requires some PNPM-specific assumptions that go beyond the generic Node.js `require()` model, and which may change in future versions of PNPM.

**Comment by octogonz** [comment]
@arikon If you're able to share a branch that repros this problem, that would give us something to test when we implement the fix.

**Comment by arikon** [comment]
My repo is private =(

**Comment by octogonz** [comment]
@willsoto provided a repro for https://github.com/microsoft/rushstack/issues/1943 that seems to be the same problem. We can use that for testing.

**Comment by octogonz** [comment]
Here's a prototype of a fix:  PR https://github.com/microsoft/rushstack/pull/1949

@arikon Can you test this out using your private repo?

To test it, you would [build the Rush Stack repo](https://rushstack.io/pages/contributing/get_started/) and then instead of running "rush deploy", invoke the `rush-lib`  engine directly like this:

```
$ node ~/my-git-stuff/rushstack/apps/rush-lib/lib/start.js deploy --overwrite
```

**Comment by octogonz** [comment]
@arikon Are you able to test this? Since it fixed @willsoto 's issue, I'd like to get it merged+published.

**Comment by arikon** [comment]
@octogonz I'll try to test it tomorrow

**Comment by arikon** [comment]
@octogonz It works in my case

**Comment by octogonz** [comment]
Thanks! I'm on a trip right now, but I'll try to get this fix released asap.

**Comment by octogonz** [comment]
This fix was released with Rush 5.27.2
